staging/src/k8s.io/legacy-cloud-providers/azure/azure_loadbalancer.go

```diff
@@ -156,6 +157,15 @@ func (az *Cloud) EnsureLoadBalancer(ctx context.Context, clusterName string, ser
 	serviceName := getServiceName(service)
 	klog.V(5).Infof("ensureloadbalancer(%s): START clusterName=%q", serviceName, clusterName)
 
+	mc := metrics.NewMetricContext("services", "ensure_loadbalancer", az.ResourceGroup, az.SubscriptionID, serviceName)
+	operationSucceeded := false
+	defer func() {
+		mc.ObserveOperation()
```

Comment:
Could we also add `operationSucceeded` in `ObserveOperation()`, e.g. we could add another function for this purpose, `ObserveOperationWithResult(operationSucceeded)`